frappe/oauth.py

```diff
@@ -164,28 +154,72 @@ def validate_code(self, client_id, code, client, request, *args, **kwargs):
 		# Validate the code belongs to the client. Add associated scopes,
 		# state and user to request.scopes and request.user.
 
-		validcodes = frappe.get_all("OAuth Authorization Code", filters={"client": client_id, "validity": "Valid"})
+		validcodes = frappe.get_all(
+			"OAuth Authorization Code",
+			filters={"client": client_id, "validity": "Valid"},
+		)
 
 		checkcodes = []
 		for vcode in validcodes:
 			checkcodes.append(vcode["name"])
 
 		if code in checkcodes:
-			request.scopes = frappe.db.get_value("OAuth Authorization Code", code, 'scopes').split(get_url_delimiter())
-			request.user = frappe.db.get_value("OAuth Authorization Code", code, 'user')
+			request.scopes = frappe.db.get_value(
+				"OAuth Authorization Code", code, "scopes"
+			).split(get_url_delimiter())
+			request.user = frappe.db.get_value("OAuth Authorization Code", code, "user")
+			code_challenge_method = frappe.db.get_value(
+				"OAuth Authorization Code", code, "code_challenge_method"
+			)
+			code_challenge = frappe.db.get_value(
+				"OAuth Authorization Code", code, "code_challenge"
+			)
+
+			if code_challenge and not request.code_verifier:
+				if frappe.db.exists("OAuth Authorization Code", code):
+					frappe.delete_doc(
+						"OAuth Authorization Code", code, ignore_permissions=True
+					)
+					frappe.db.commit()
```

Comment:
This could be a `delete_doc_if_exists` wrapper call instead. I'm not sure if it takes the `ignore_permissions` parameter though.

Reply:
It doesn't take `ignore_permissions`, this is workaround.

Reply:
Alright, this is fine then